src/config.ts

```diff
@@ -0,0 +1,60 @@
+import { GraphQLFieldConfig, GraphQLInterfaceType } from 'graphql';
+// eslint-disable-next-line max-classes-per-file
+import { fromGlobalId, nodeDefinitions } from 'graphql-relay';
+import invariant from 'invariant';
+
+import Resource from './resources/Resource';
+import { Context } from './types/Context';
+// eslint-disable-next-line import/no-cycle
+import NodeType from './types/NodeType';
+import ResourceManager from './types/ResourceManager';
+
+type Config = {
+  nodeField: GraphQLFieldConfig<any, Context>;
+  nodesField: GraphQLFieldConfig<any, Context>;
+  localIdFieldMode: 'include' | 'omit' | 'deprecated';
+  resourceManager: ResourceManager;
+  typesManager: Map<string, NodeType<Resource<Context>, any>>;
+  nodeInterface: GraphQLInterfaceType;
+};
+
+export const config: Readonly<Config> = {} as any;
+
+function createConfig({
+  localIdFieldMode = 'omit',
+}: Pick<Config, 'localIdFieldMode'>): Config {
+  const typesManager = new Map<string, NodeType<Resource<Context>, any>>();
+
+  const { nodeInterface, nodeField, nodesField } = nodeDefinitions<Context>(
+    async (globalId, context) => {
+      const { type, id } = fromGlobalId(globalId);
+      const resolvesType = typesManager.get(type);
+
+      invariant(resolvesType, 'There is no matching type');
+      const item = await resolvesType.getResource(context).get(id);
+
+      if (!item) return null;
+
+      return { $type: type, ...item };
+    },
+
+    obj => typesManager.get(obj.$type),
+  );
+
+  return {
+    resourceManager: new ResourceManager(),
+    typesManager,
+    localIdFieldMode,
+    nodeInterface,
+    nodeField,
+    nodesField,
+  };
+}
+
+export function needsSetup() {
+  return Object.keys(config).length === 0;
+}
```

Comment:
this seems slightly awkward. could you just define e.g. `config` as `Config | null`, and have a `getConfig` method that always does the check and returns `Config`?

Reply:
sucks to call getConfig all the time

Reply:
I could you make config a proxy that checks that though 😬

Reply:
i mean how many times are you accessing the config? it's not that much. and it's never in user code

Reply:
true